test/test.config.js

```diff
@@ -7,29 +7,39 @@ import path from 'node:path';
 
 const __dirname = path.dirname(fileURLToPath(import.meta.url));
 
+const convertToBoolean = (envVariable = '') => {
+  if(/^(1|true)$/.test(envVariable)) {
+    return true;
+  }
+  return false;
+};
+
+const {connectOptions} = config.mongodb;
+
 // MongoDB
 config.mongodb.name = 'bedrock_mongodb_test';
 config.mongodb.host = process.env.MONGODB_HOST || 'localhost';
 config.mongodb.port = process.env.MONGODB_PORT || 27017;
-
-if(process.env.MONGODB_USERNAME && process.env.MONGODB_PASSWORD) {
-  config.mongodb.username = process.env.MONGODB_USERNAME;
-  config.mongodb.password = process.env.MONGODB_PASSWORD;
-  const {connectOptions} = config.mongodb;
-  // process.env.MONGODB_SSL should be 1 to test with SSL
-  connectOptions.ssl = Boolean(process.env.MONGODB_SSL) || false;
-  connectOptions.authSource = process.env.MONGODB_AUTHSOURCE || 'admin';
-  // this can safely be undefined
-  connectOptions.replicaSet = process.env.MONGODB_REPLICASET;
-  console.log(
-    'TESTING WITH AUTH ', {
-      username: config.mongodb.username,
-      authSource: connectOptions.authSource,
-      password: Boolean(config.mongodb.password),
-      ssl: connectOptions.ssl,
-      replicaSet: connectOptions.replicaSet
-    });
+// set the env variable to 1 or true to make these true
+// set the env variable to anything else to make them false
+if(process.env.MONGODB_SSL) {
+  connectOptions.ssl = convertToBoolean(process.env.MONGODB_SSL);
+}
+if(process.env.MONGODB_TLS) {
+  connectOptions.tls = convertToBoolean(process.env.MONGODB_TLS);
 }
+if(process.env.MONGODB_AUTHSOURCE) {
+  connectOptions.authSource = process.env.MONGODB_AUTHSOURCE;
+}
+
+// used for testing url only connections
+config.mongodb.url = process.env.MONGODB_URL;
+// this can safely be undefined
+connectOptions.replicaSet = process.env.MONGODB_REPLICASET;
+
+config.mongodb.username = process.env.MONGODB_USERNAME;
+config.mongodb.password = process.env.MONGODB_PASSWORD;
```

Comment:
Same question here ... does this just continue to work for devs running things locally? It seems like it would break.

Reply:
It just seems like these should only be set if the environment variable is also set. Otherwise someone modifying their local test config will be confused when it doesn't work. So please add `if` conditionals here -- we shouldn't override something someone has set in the config unless the environment variable is actually set.

Reply:
Good idea and addressed here: https://github.com/digitalbazaar/bedrock-mongodb/pull/89/commits/3f4adaf8125c1a91910df261065578b738654944